boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,933 @@
+{
+    return HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin);
+}
+
+void Io_Efuse_DelatchAux1Aux2Faults(void)
+{
+    // Delatch the latchable faults by alternating the inputs high-low-high
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+}
+
+float Io_Efuse_GetAux1Current(void)
+{
+    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
+            AUX1_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
+    {
+        return NAN;
+    }
+
+    if (HAL_GPIO_ReadPin(
+            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return Io_CurrentSense_GetAux1Current();
+    }
+
+    return NAN;
+}
+
+float Io_Efuse_GetAux2Current(void)
+{
+    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
+            AUX2_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
+    {
+        return NAN;
+    }
+
+    if (HAL_GPIO_ReadPin(
+            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return Io_CurrentSense_GetAux2Current();
+    }
+
+    return NAN;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection)
+{
+    uint16_t reg_val = 0x0000;
+
+    // Read original content of GCR Register
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, &reg_val));
+
+    CLEAR_BIT(reg_val, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(reg_val, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, reg_val));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_Aux1Aux2ConfigureEfuse(void)
+{
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2ExitFailSafeMode());
+
+    // Global config register
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG));
+
+    // Channel 0 config registers
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_0_ADDR, RETRY_CONFIG));
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_0_ADDR, CONFR_CONFIG));
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_0_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG));
+
+    // Channel 1 config registers
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_1_ADDR, RETRY_CONFIG));
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_1_ADDR, CONFR_CONFIG));
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_1_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG));
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(void)
+{
+    // Set WDIN bit for next write
+    // 1_1_00000_00000_0000
```

Comment:
What are these bits used for? Are these the bits we need to transfer to set WDIN for next write?

Reply:
The 15th bit is the WDIN bit that needs to be set to exit fail-safe mode. The 14th bit is the parity bit which will be one in this case since the parity is odd.

Reply:
Actually can you add your comment above to the comment in this function?

Reply:
Just reading the code again I just saw a bunch of 1's and 0's and didn't know why it was relevant lol

Reply:
haha ya added a comment, let me know if it makes more sense